src/components/PopupWithForm.js

```diff
@@ -28,6 +28,7 @@ export default class PopupWithForm extends Popup {
       evt.preventDefault();
       this._handleFormSubmit(this._getInputValues());
       this._popupForm.reset();
+      this._submitButton.textContent = "Saving...";
       this.close();
```

Comment:
Needs correction: Сlose modal in the case of successful api response in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.